### PR TITLE
CDAP 19697 Feature - Import Data

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
Feature - Import Data

## Description
Summary of changes
This PR contains the Import Data feature in the Connection List Page

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19697

## Files to be Reviewed
Please review the following files as part of this feature
1. app/cdap/components/ImportDataset/
2. app/cdap/text/text-en.yaml
3. src/e2e-test/

The other files are requested to be ignored as they either contain one of the following
1. Quick integrations for connecting the current feature with the Pages.
2. Another feature change that will have individual feature branches. Once that feature branch gets merged, then we won't see many changes w.r.t. these files.
4. Quick fixes for failing test cases, as they are not part of the current feature but may arise due to merging and conflict resolutions.

## Test Plan
1. FE_09_072 - Verify whether the import data text is displayed. - Pass
5. FE_09_073 - Verify by mouse hovering for the "Download" icon - Pass
6. FE_09_074 - Verify by clicking on the "Download" icon - Pass
7. FE_09_075 - Verify the direction of the "Download icon" by clicking. - Pass
8. FE_09_076 - Verify the "Import data text link" by mouse hovering. - Pass
9. FE_09_077 - Verify by clicking on the "Import data text link" icon - Pass
10. FE_09_078 - Verify the direction of the "Import data text link" by clicking. - Pass
11. FE_09_079 - Verify the Paddings & alignments between the Download icon and Import data text link as per design or not. - Pass
12. FE_09_080 - Verify whether the "Import Data" text is displaying as a title or not. - Pass
13. FE_09_081 - Verify by clicking on the close icon of the "Import Data" popup screen - Pass
14. FE_09_082 - Verify whether the drag & drop box is available as per design or not. - Pass
15. FE_09_083 - Verify by clicking on the drag & drop box. - Pass
16. FE_09_084 - Verify the redirection after clicking on the drag & drop box. - Pass
17. FE_09_085 - Verify the infographic (image) is displayed or not when we don't have any selected file. - Pass
18. FE_09_086 - Verify the infographic (image) is displayed  as per design or not - Pass
19. FE_09_087 - Verify the text "Drag & drop a file to be uploaded or click here to select a file from your computer" below whether the infographic is displayed as per Figma or not. - Pass
20. FE_09_088 - Verify the Paddings & alignment of the Infographic (image) and text inside the Drag & drag Box as per design - Pass
21. FE_09_089 - Verify the info icon beside the "Max file size" text by mouse hovering - Pass
22. FE_09_090 - Verify the text "Max file size: 10 MB, Supported file formats: CSV, JSON" is displayed as per Figma or not. - Pass
23. FE_09_091 - Verify by uploading the different file formats in the drag & drop box. - Pass
24. FE_09_092 - Verify by dragging & droping the file into the window. - Pass
25. FE_09_093 - Verify the actions after uploading a file - Pass
26. FE_09_094 - Verify after dragging & dropping a file whether the local machine window is closed or not. - Pass
27. FE_09_095 - Verify by mouse hovering on the delete icon. - Pass.
28. FE_09_096 - Verify by clicking on the delete icon. - Pass.
29. FE_09_097 - Verify the redirection of the screen after deleting the selected list.  - Pass.
30. FE_09_098 - Verify whether the "Wrangle" button is displayed or not after uploading a file. - Pass
31. FE_09_099 - Verify By mouse hovering on the "Wrangle" button after uploading any file. - Pass
32. FE_09_100 - Verify By clicking on the "Wrangle" button - Pass
33. FE_09_101 - Verify By Submitting the "Wrangle" button - Pass
34. FE_09_102 - Verify the Redirection after Submitting the "Wrangle" button - Pass

Please find the list of Testcases in the following sheet
https://docs.google.com/spreadsheets/d/1lchTKplHp2WBryRCHSvxivdNzbmgcL8pi1GE9KmoK8Y/edit#gid=906164546

## Screenshots
Screenshot 2022-11-10 at 10 00 40 PM
Screenshot 2022-11-10 at 10 00 48 PM
Screenshot 2022-11-10 at 10 00 57 PM
Screenshot 2022-11-10 at 10 01 10 PM
Screenshot 2022-11-10 at 10 01 22 PM

### CODE REVIEW

File: app/cdap/components/ImportDataset/Components/DragAndDrop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      ) : (
+        <Box>
+          <Box className={classes.FlexFile}>
+            <Typography className={classes.fileNameText} data-testid="file-name" component="span">
+              {file.name}
+            </Typography>
+            <Box
+              className={classes.delete_cursor_pointer}
+              data-testid="delete-svg"
+              onClick={handleRemoveFile}
```
Comment (njbriggs) [inline app/cdap/components/ImportDataset/Components/DragAndDrop/index.tsx]: For accessibility purposes, do not add click handlers to `div`s/`Box`es. Use a `Button`.

File: app/cdap/components/ImportDataset/IconStore/DeleteSVG.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+
+export const DeleteSVG = () => {
```
Comment (njbriggs) [inline app/cdap/components/ImportDataset/IconStore/DeleteSVG.tsx]: Please use the Delete/trash can icon from Mui

File: app/cdap/components/ImportDataset/IconStore/InfoIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+
+export const InfoIcon = () => {
```
Comment (njbriggs) [inline app/cdap/components/ImportDataset/IconStore/InfoIcon.tsx]: Please use the Info icon from Mui

File: app/cdap/components/ImportDataset/Components/DragAndDrop/index.tsx (hunk 1 above)
Comment (dlakshmi123) [reply app/cdap/components/ImportDataset/Components/DragAndDrop/index.tsx]: removed onClick to Box

File: app/cdap/components/ImportDataset/IconStore/DeleteSVG.tsx (hunk 2 above)
Comment (dlakshmi123) [reply app/cdap/components/ImportDataset/IconStore/DeleteSVG.tsx]: Fixed, added MUI Icon

File: app/cdap/components/ImportDataset/IconStore/InfoIcon.tsx (hunk 3 above)
Comment (dlakshmi123) [reply app/cdap/components/ImportDataset/IconStore/InfoIcon.tsx]: Fixed, added MUI Icon

**Pull request closed without merging**